engine/src/mac-core.mm

```diff
@@ -669,6 +669,7 @@ void MCPlatformSetSystemProperty(MCPlatformSystemProperty p_property, MCPlatform
 
 static MCModalSession *s_modal_sessions = nil;
 static uindex_t s_modal_session_count = 0;
+void (^s_modal_session_cleanup)(void) = nil;
```

Comment:
So using a closure here is clever - however the fact it is nested makes it really difficult to understand (I think I get what is going on but probably wouldn't if I looked at this code again).

Can I suggest we change s_modal_sessions to be an MCAutoArray<MCModalSession> (to get rid of the explicit management of the vector) - then add an
```
MCAutoArray<MCModalSession> s_modal_sessions_pending_cleanup
```
So we replace what appears to be just a chain of cleanup closures with explicit iteration through an array?